docs/core/entity/climate.md

```diff
@@ -142,6 +142,52 @@ class MyClimateEntity(ClimateEntity):
         """Set new target hvac mode."""
 ```
 
+### Turn on
+
+```python
+class MyClimateEntity(ClimateEntity):
+    # Implement one of these methods.
+    # The `turn_on` method should set `hvac_mode` to any other than
+    # `HVACMode.OFF` by optimistically setting it from the service handler
+    # or with the next state update
+
+    def turn_on(self):
+        """Turn the entity on."""
+
+    async def async_turn_on(self):
+        """Turn the entity on."""
+```
+
+### Turn off
+
+```python
+class MyClimateEntity(ClimateEntity):
+    # Implement one of these methods.
+    # The `turn_off` method should set `hvac_mode` to `HVACMode.OFF` by
+    # optimistically setting it from the service handler or with the next state update
+
+    def turn_off(self):
+        """Turn the entity off."""
+
+    async def async_turn_off(self):
+        """Turn the entity off."""
+```
+
+### Toggle
+
+```python
+class MyClimateEntity(ClimateEntity):
+    # Implement one of these methods.
+    # The `toggle` method should set `hvac_mode` to the right `HVACMode` by
+    # optimistically setting it from the service handler or with the next state update
+
+    def toggle(self):
+        """Toggle the entity."""
+
+    async def async_toggle(self):
+        """Toggle the entity."""
+```
```

Comment:
We should explain it's not mandatory to implement toggle, if not implemented the base entity's will call turn_on or turn_off according to the current HVAC mode.